[13.0][ADD] account_move_line_cumulated_balance: Show field cumulated balance

cesar-tecnativa opened the pull request:
@Tecnativa TT37842

cesar-tecnativa committed:
[ADD] account_move_line_cumulated_balance: cumulated balance

Co-Authored-By: Pedro M. Baeza <pedro.baeza@tecnativa.com>

cesar-tecnativa commented:
@pedrobaeza could you please review?

cesar-tecnativa commented:
@chienandalu Please, could you review?

cesar-tecnativa commented:
@victoralmau could you please review this?

cesar-tecnativa commented:
@LudLaf could you review this?

pedrobaeza commented:
@LudLaf please review it functionally

LudLaf commented:
> @LudLaf please review it functionally

Just tested. Functionality is working.

pedrobaeza commented:
@LudLaf do the proper review with GitHub tools

LudLaf reviewed (commented):
@cesar-tecnativa something has changed. When trying to test this module, it shows the following:

Error:


Traceback:
Error: fieldsViews is undefined
_generateActionViews/<@http://oca-account-financial-tools-13-0-pr1438-d1a1c2932c24.runboat.odoo-community.org/web/static/src/js/chrome/action_manager_act_window.js:355:30
_.forEach@http://oca-account-financial-tools-13-0-pr1438-d1a1c2932c24.runboat.odoo-community.org/web/static/lib/underscore/underscore.js:145:17
_generateActionViews@http://oca-account-financial-tools-13-0-pr1438-d1a1c2932c24.runboat.odoo-community.org/web/static/src/js/chrome/action_manager_act_window.js:353:11
_executeWindowAction/<@http://oca-account-financial-tools-13-0-pr1438-d1a1c2932c24.runboat.odoo-community.org/web/static/src/js/chrome/action_manager_act_window.js:258:30

chienandalu commented:
> @cesar-tecnativa something has changed. When trying to test this module, it shows the following:


Please provide steps to reproduce the issue.

LudLaf commented:
> > @cesar-tecnativa something has changed. When trying to test this module, it shows the following:
> 
> Please provide steps to reproduce the issue.
![Peek 2022-08-11 10-33](https://user-images.githubusercontent.com/4016592/184095484-e1d7dc9f-5717-4caf-9d55-7038e913005d.gif)

cesar-tecnativa commented:
> > > @cesar-tecnativa something has changed. When trying to test this module, it shows the following:
> > 
> > 
> > Please provide steps to reproduce the issue.
> > ![Peek 2022-08-11 10-33](https://user-images.githubusercontent.com/4016592/184095484-e1d7dc9f-5717-4caf-9d55-7038e913005d.gif)
> >       
> >         [
> >           
> >             ![Peek 2022-08-11 10-33](https://user-images.githubusercontent.com/4016592/184095484-e1d7dc9f-5717-4caf-9d55-7038e913005d.gif)
> >           
> >         ](https://user-images.githubusercontent.com/4016592/184095484-e1d7dc9f-5717-4caf-9d55-7038e913005d.gif)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://user-images.githubusercontent.com/4016592/184095484-e1d7dc9f-5717-4caf-9d55-7038e913005d.gif)
solved

LudLaf reviewed (commented):
Module is shown as "To be installed", but doesn't allow you to keep installing nor using it

![Peek 2022-08-11 12-29](https://user-images.githubusercontent.com/4016592/184114908-a52de4ca-51a0-4d3c-b21d-96ab707d0b11.gif)
.

cesar-tecnativa commented:
new changes are done

LudLaf commented:
> 



> new changes are done

Just tried to open the invoicing app, but didn't work.


---
Error:


Traceback:
Error: fieldsViews is undefined
_generateActionViews/<@http://oca-account-financial-tools-13-0-pr1438-82728466fadf.runboat.odoo-community.org/web/static/src/js/chrome/action_manager_act_window.js:355:30
_.forEach@http://oca-account-financial-tools-13-0-pr1438-82728466fadf.runboat.odoo-community.org/web/static/lib/underscore/underscore.js:145:17
_generateActionViews@http://oca-account-financial-tools-13-0-pr1438-82728466fadf.runboat.odoo-community.org/web/static/src/js/chrome/action_manager_act_window.js:353:11
_executeWindowAction/<@http://oca-account-financial-tools-13-0-pr1438-82728466fadf.runboat.odoo-community.org/web/static/src/js/chrome/action_manager_act_window.js:258:3
![Peek 2022-08-15 10-22](https://user-images.githubusercontent.com/4016592/184601695-3929ceca-590f-4990-a583-88b29e0f23f5.gif)
0

LudLaf reviewed (changes requested):
Not working on runboat nor when trying to install it in local, so it's not possible to test it functionally. Please check previous review.

chienandalu commented on account_move_line_cumulated_balance/models/account_move_line.py:
```diff
@@ -0,0 +1,85 @@
+    @api.model
+    def search_read(self, domain=None, fields=None, offset=0, limit=None, order=None):
+        # Make an explicit order because we will need to reverse it
+        order = (order or self._order) + ", id"
```
Shouldn the order key go as something like `", id DESC"` or `", id ASC"`. What if `id` is already in the order string?

chienandalu commented:
> Not working on runboat nor when trying to install it in local, so it's not possible to test it functionally. Please check previous review.

I just tested it on runbot. I had to install `account_lock_date_update` which was missing on some view for any reason.

chienandalu commented on account_move_line_cumulated_balance/models/account_move_line.py:
```diff
@@ -0,0 +1,73 @@
+            FROM account_move_line
+            LEFT JOIN account_move on account_move_line.move_id = account_move.id
+            WHERE
+            account_move.state = 'posted'
```
Why not filtering the `self` recordset `ids`. This query will be increasingly heavy, isn't it?

pedrobaeza committed:
[FIX] account_move_line_cumulated_balance: Absolute SQL strategy

As other partial ones doesn't fully work. The performance is OK enough
for a big amount of records (~2 s for 25k records on the same account),
so we accept the trade-off for having correct data.

pedrobaeza commented:
@chienandalu you can check now. Data is computed in a absolute way record per record with an aggregated SQL, but it's efficient enough on real databases, and it's the only way to get proper data no matter done filters.

chienandalu commented:
> @chienandalu you can check now. Data is computed in a absolute way record per record with an aggregated SQL, but it's efficient enough on real databases, and it's the only way to get proper data no matter done filters.

And won't it produce tons of queries?

pedrobaeza commented:
As you visualize a limited number of records in these views, that would be the number of queries. It can't be avoided due to the nature of the need.

chienandalu reviewed (approved):
:+1:

pedrobaeza commented:
/ocabot merge nobump
